tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3705,8 +3889,8 @@ private[optimizer] object OptimizerCore {
     }
 
     def apply(binding: PreTransBinding,
-        result: PreTransLocalDef): PreTransform = {
-      if (result.contains(binding.localDef)) {
+        result: PreTransResult): PreTransform = {
+      if (result.contains(binding.localDef) || binding.isAlreadyUsed) {
```

Comment:
> After that, the top-level PreTransBinaryOp does contains(binding.localDef), but it used the binding nevertheless.

If that was supposed to be "does *not* `contain(binding.localDef)`, I understand and this makes sense. :)